examples/larger-examples/datalog/single-source-shortest-distance.flix

```diff
@@ -0,0 +1,181 @@
+mod ShortestDistance {
+
+    use Down.Down;
+
+    ///
+    /// Returns the shortest distance from `src` to every other node in the
+    /// weighted directed graph `g`.
+    ///
+    pub def distancesFrom(src: t, g: m[(t, t, Int32)]): Map[t, Int32] with Foldable[m], Order[t] = {
+        let edges = inject g into Edge;
+        let dists = #{
+            Dist(src; Down.Down(0)).
+            Dist(y; d + Down.Down(w)) :- Dist(x; d), Edge(x, y, w).
+        };
+        let mapping = #{
+            Mapping(x, destructDown(d)) :- fix Dist(x; d).
```

Comment:
Cute, kind-of. Gave me an idea. I shall return...

Reply:
This is a copy of the std lib :)

Reply:
So any comments here could be applied to the library too

Reply:
Check out https://github.com/flix/flix/pull/7216 for a simple solution. (Needs to be merged first).

Reply:
Means we can drop Mapping and just use coerce.